### PR TITLE
Add elevator questions

### PR DESCRIPTION
**Pull request by wielandb**
This PR adds some questions that relate to the recent "Evelator dimensions" proposal.

These questions are:
- What shape does this elevator have?
- What's the diameter of this elevator?
- How tall is the cabin of this elevator?
- How wide is the door of this elevator?
- How tall is the door of this elevator?

It modifies the questions
- How wide is this elevator?
- How deep is this elevator?
to only be asked if the elevator has a rectangular shape

It does not ask for any `max*:physical=*`values as this is currently not possible with this app.

Currently missing:
- Example images

### CODE REVIEW

File: assets/question_catalog/locales/en.arb [hunk 1]
```diff
@@ -617,5 +617,43 @@
...
+  "@elevatorHeightPlaceholder": {},
+  "elevatorDiameterName": "Elevator diameter",
+  "@elevatorDiameterName": {},
+  "elevatorDiameterText": "What's the diameter of this elevator?",
+  "@elevatorDiameterText": {},
+  "elevatorDiameterPlaceholder": "Diameter of the cabin in meters",
+  "@elevatorDiameterPlaceholder": {},
+  "elevatorDoorWidthName": "Elevator door width",
+  "@elevatorDoorWidthName": {},
+  "elevatorDoorWidthText": "How wide is the door of this elevator?",
```
Comment (7h30n3) [inline assets/question_catalog/locales/en.arb:645]: To be more in line with existing questions: "What is the width of the elevator door?

File: assets/question_catalog/locales/en.arb [hunk 2]
```diff
@@ -617,5 +617,43 @@
...
+  "@elevatorDoorWidthText": {},
+  "elevatorDoorWidthDescription": "This means the inner clearance from one side of the door to the other, not the width of the frame"
+  "@elevatorDoorWidthDescription": {},
+  "elevatorDoorWidthPlaceholder": "Clearance in meters",
+  "@elevatorDoorWidthPlaceholder": {},
+  "elevatorDoorHeightName": "Elevator door height",
+  "@elevatorDoorHeightName": {},
+  "elevatorDoorHeightText": "How tall is the door of this elevator?",
```
Comment (7h30n3) [inline assets/question_catalog/locales/en.arb:653]: To be more in line with existing questions: "What is the height of the elevator door?"

File: assets/question_catalog/locales/en.arb [hunk 3]
```diff
@@ -617,5 +617,43 @@
...
+  "@elevatorDoorHeightText": {},
+  "elevatorDoorHeightDescription": "This means the inner clearance from the bottom of the door to the top, not the height of the frame"
+  "@elevatorDoorHeightDescription": {},
+  "elevatorDoorHeightPlaceholder": "Clearance in meters",
```
Comment (7h30n3) [inline assets/question_catalog/locales/en.arb:657]: Just "Height".

**Review by 7h30n3 — CHANGES_REQUESTED** [review]

File: assets/question_catalog/definition.json [hunk 4]
```diff
@@ -4116,5 +4118,179 @@
...
+      "conditions": [
+         {
+            "osm_tags": {
+               "highway": "elevator",
+               "shape": false
+            },
+            "osm_element": "Node"
+         }
+      ]
+   },
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:4159]: For the first PR please omit the circular elevator case.
I'm not sure if we really want this in OpenStop due to the requirement of the `shape=rectangular` tag.

File: assets/question_catalog/definition.json [hunk 5]
```diff
@@ -4116,5 +4118,179 @@
...
+      "question": {
+         "name": "@elevatorHeightName",
+         "text": "@elevatorHeightText"
+      },
+      "answer": {
+         "type": "Number",
+         "input": {
+            "placeholder": "@elevatorHeightPlaceholder",
+            "decimals": 1,
+            "min": 1,
```
[the remark this answers is not in the document]
Comment (Robbendebiene) [reply assets/question_catalog/definition.json:4170]: How about 1.5m / 150cm?

File: assets/question_catalog/definition.json [hunk 6]
```diff
@@ -4116,5 +4118,179 @@
...
+            "max": 4,
+            "unit": "@unitMeter"
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:4172]: The units in the existing elevator question are in centimetres so please align the new questions accordingly.

File: assets/question_catalog/definition.json [hunk 7]
```diff
@@ -4116,5 +4118,179 @@
...
+                  {
+                    "osm_tags": {
+                      "indoor": "door"
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+      ]
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:4255]: What is the idea of this? To exclude door nodes PART of an elevator way or door nodes connected to an elevator via a footway?

**Review by Robbendebiene — CHANGES_REQUESTED** [review]

File: assets/question_catalog/definition.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (wielandb) [reply assets/question_catalog/definition.json:4170]: Used 1.5

File: assets/question_catalog/locales/en.arb [hunk 8]
```diff
@@ -617,5 +617,43 @@
...
+  "@unitKilogram": {},
+  "elevatorShapeName": "Elevator shape",
+  "@elevatorShapeName": {},
+  "elevatorShapeText": "What shape does this elevator have?",
+  "@elevatorShapeText": {},
+  "elevatorShapeRectangularOption": "Rectangular"
+  "@elevatorShapeRectangularOption": {},
+  "elevatorShapeCircularOption": "Circular",
+  "@elevatorShapeCircularOption": {},
+  "elevatorShapeCustomOption": "Something else",
```
[the remark this answers is not in the document]
Comment (wielandb) [reply assets/question_catalog/locales/en.arb:629]: Used Complex

File: assets/question_catalog/definition.json (hunk 4 above)
Comment (wielandb) [reply assets/question_catalog/definition.json:4159]: Could you explain further? What problems could arise?

File: assets/question_catalog/definition.json (hunk 7 above)
Comment (wielandb) [reply assets/question_catalog/definition.json]: Connected via footway

**Comment by wielandb** [comment]
The JSON validation fails.

I don't really understand what's wrong about
```
            "max": 4,
            "unit": "@unitCentimeter"
         },
         "constructor": {
            "diameter": ["CONCAT", "$input", "m"]
         }
      },
```

File: assets/question_catalog/locales/en.arb [hunk 9]
```diff
@@ -617,5 +617,43 @@
...
+  "@elevatorDoorWidthText": {},
+  "elevatorDoorWidthDescription": "This means the inner clearance from one side of the door to the other, not the width of the frame",
+  "@elevatorDoorWidthDescription": {},
+  "elevatorDoorWidthPlaceholder": "Clearance",
+  "@elevatorDoorWidthPlaceholder": {},
+  "elevatorDoorHeightName": "Elevator door height",
+  "@elevatorDoorHeightName": {},
+  "elevatorDoorHeightText": "What is the height of the elevator door?",
+  "@elevatorDoorHeightText": {},
+  "elevatorDoorHeightDescription": "This means the inner clearance from the bottom of the door to the top, not the height of the frame"
```
Comment (Robbendebiene) [inline assets/question_catalog/locales/en.arb:655]: The clearance height is the inner measurement from the bottom of the door to the top, not the outer height of the door frame.

File: assets/question_catalog/locales/en.arb [hunk 10]
```diff
@@ -617,5 +617,43 @@
...
+  "@elevatorDiameterName": {},
+  "elevatorDiameterText": "What is the diameter of this elevator?",
+  "@elevatorDiameterText": {},
+  "elevatorDiameterPlaceholder": "Diameter of the cabin",
+  "@elevatorDiameterPlaceholder": {},
+  "elevatorDoorWidthName": "Elevator door width",
+  "@elevatorDoorWidthName": {},
+  "elevatorDoorWidthText": "What is the width of the elevator door?",
+  "@elevatorDoorWidthText": {},
+  "elevatorDoorWidthDescription": "This means the inner clearance from one side of the door to the other, not the width of the frame",
```
Comment (Robbendebiene) [inline assets/question_catalog/locales/en.arb:647]: The clearance width is the inner measurement from one side of the door to the other, not the outer width of the door frame.

**Review by Robbendebiene — CHANGES_REQUESTED** [review]

File: assets/question_catalog/definition.json (hunk 4 above)
Comment (wielandb) [reply assets/question_catalog/definition.json:4159]: - Right, but how do we handle that?
- Do you think resolving https://github.com/OPENER-next/OpenStop/issues/270 would address this?
- I personally think the diameter of a perfectly round elevator is not difficult to measure. o.o

Comment (wielandb) [reply assets/question_catalog/definition.json:4159]: There is actually some kind of precedent: You can tag `roof:shape=many` for roofs. 

So, your preferred option would be that we (for now) ignore everything that does not concern rectangular elevators?

File: assets/question_catalog/locales/de.arb [hunk 11]
```diff
@@ -315,7 +315,10 @@
   "cycleBarrierWidthDescription": "Die Durchgangsbreite, auch lichte Breite genannt, beschreibt die maximal zulässige Breite, um eine Sperre passieren zu können.",
   "@cycleBarrierWidthDescription": {},
   "cycleBarrierWidthPlaceholder": "Breite",
-  "@cycleBarrierWidthPlaceholder": {},
+  "@cycleBarrierWidthPlaceho,
+         {
+
+          }lder": {},
```
[the remark this answers is not in the document]
Comment (wielandb) [reply assets/question_catalog/locales/de.arb:321]: Da heck happend there? 🙈

File: assets/question_catalog/definition.json [hunk 12]
```diff
@@ -4116,5 +4118,177 @@
...
+         "input": {
+            "placeholder": "@elevatorHeightPlaceholder",
+            "decimals": 1,
+            "min": 1.5,
+            "max": 4,
+            "unit": "@unitCentimeter"
+         },
+         "constructor": {
+            "height": ["$input"]
+         }
```
[the remark this answers is not in the document]
Comment (wielandb) [reply assets/question_catalog/definition.json]: I think I fixed everything now!

File: assets/question_catalog/definition.json (hunk 7 above)
Comment (wielandb) [reply assets/question_catalog/definition.json]: Keeping it as it is

File: assets/question_catalog/locales/de.arb [hunk 13]
```diff
@@ -617,5 +617,43 @@
...
+  "@elevatorDiameterName": {},
+  "elevatorDiameterText": "Welchen Durchmesser hat die Kabine des Fahrstuhls?",
+  "@elevatorDiameterText": {},
+  "elevatorDiameterPlaceholder": "Durchmesser der Kabine",
+  "@elevatorDiameterPlaceholder": {},
+  "elevatorDoorWidthName": "Breite der Fahrstuhltür",
+  "@elevatorDoorWidthName": {},
+  "elevatorDoorWidthText": "Wie breit ist die Tür dieses Fahrstuhls?",
+  "@elevatorDoorWidthText": {},
+  "elevatorDoorWidthDescription": "Gemeint ist die lichte Breite der Tür, also die Breite der Türöffnung, nicht die Breite des Türrahmens.",
```
Comment (Robbendebiene) [inline assets/question_catalog/locales/de.arb:647]: Please adjust according to the English translation.

File: assets/question_catalog/locales/de.arb [hunk 14]
```diff
@@ -617,5 +617,43 @@
...
+  "@elevatorDoorWidthDescription": {},
+  "elevatorDoorWidthPlaceholder": "Breite",
+  "@elevatorDoorWidthPlaceholder": {},
+  "elevatorDoorHeightName": "Höhe der Fahrstuhltür",
+  "@elevatorDoorHeightName": {},
+  "elevatorDoorHeightText": "Wie hoch ist die Tür dieses Fahrstuhls?",
+  "@elevatorDoorHeightText": {},
+  "elevatorDoorHeightDescription": "Gemeint ist die lichte Höhe der Tür, also die Höhe der Türöffnung, nicht die Höhe des Türrahmens.",
```
Comment (Robbendebiene) [inline assets/question_catalog/locales/de.arb:655]: Please adjust according to the English translation.

File: assets/question_catalog/locales/en.arb [hunk 15]
```diff
@@ -617,5 +617,43 @@
...
+  "elevatorShapeName": "Elevator shape",
+  "@elevatorShapeName": {},
+  "elevatorShapeText": "What is the shape of this elevator?",
+  "@elevatorShapeText": {},
+  "elevatorShapeRectangularOption": "Rectangular",
+  "@elevatorShapeRectangularOption": {},
+  "elevatorShapeCircularOption": "Round",
+  "@elevatorShapeCircularOption": {},
+  "elevatorShapeCustomOption": "Complex",
+  "@elevatorShapeCustomOption": {},
```
Comment (Robbendebiene) [inline assets/question_catalog/locales/en.arb:630]: Can be removed

**Review by Robbendebiene — CHANGES_REQUESTED** [review]

File: assets/question_catalog/definition.json [hunk 16]
```diff
@@ -4116,5 +4122,168 @@
...
+               "highway": "elevator",
+               "door:width": false
+            },
+            "!parent": [
+               {
+                 "child": [
+                   {
+                     "osm_tags": {
+                       "indoor": "door"
+                     }
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:4233]: The conditions for the elevator door height are a little tighter which I prefer. Please use the same conditions for the door width.

File: assets/question_catalog/definition.json [hunk 17]
```diff
@@ -4116,5 +4122,168 @@
...
+                      "indoor": "door",
+                      "door":"sliding",
+                      "automatic_door": ["yes", "button"]
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+      ],
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:4286]: Shouldn't we have `"osm_element": "Node"` as an additional condition for both questions?

File: assets/question_catalog/definition.json [hunk 18]
```diff
@@ -4116,5 +4122,168 @@
...
+      "$comment": "The condition is trying to make sure that there is no separately mapped elevator door by excluding the question if another door with elevator-door-like properties is on the same footway as the elevator."
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:4287]: Please also copy/paste the comment to the door width question.

**Review by Robbendebiene — CHANGES_REQUESTED** [review]

File: assets/question_catalog/definition.json (hunk 17 above)
Comment (Robbendebiene) [reply assets/question_catalog/definition.json:4286]: We decided to ask this question for Nodes AND ClosedWays but add an additional condition for the ClosedWay case to check whether a door node is already a child of the elevator.

File: assets/question_catalog/definition.json [hunk 19]
```diff
@@ -4269,8 +4281,10 @@
                "highway": "elevator",
                "door:height": false
             },
+            "osm_element": ["Node", "ClosedWay"],
             "!parent": [
               {
+                "osm_element": ["OpenWay", "ClosedWay"],
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:4287]: I think it makes sense to only look for an `OpenWay` here.

```suggestion
                "osm_element": "OpenWay",
```

File: assets/question_catalog/definition.json [hunk 20]
```diff
@@ -4224,19 +4224,31 @@
...
-                       "indoor": "door"
-                     }
-                   }
-                 ]
-               }
-             ]
-         }
-      ]
+              {
+                "osm_element": ["OpenWay", "ClosedWay"],
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:4230]: I think it makes sense to only look for an `OpenWay` here.

```suggestion
                "osm_element": "OpenWay",
```

**Review by Robbendebiene — CHANGES_REQUESTED** [review]
Move all new elevator questions right below the elevator width/length questions.

**Pull request merged**